examples/routingmanagerd/CMakeLists.txt

```diff
@@ -2,22 +2,31 @@
 # terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain
 # one at http://mozilla.org/MPL/2.0/.
 
+cmake_minimum_required(VERSION 3.15)
+
+
+option(VSOMEIP_INSTALL_ROUTINGMANAGERD "Whether or not to install the routing manager daemon.")
+
 # Daemon
-add_executable(routingmanagerd routingmanagerd.cpp)
+add_executable(routingmanagerd)
+target_sources(routingmanagerd PRIVATE routingmanagerd.cpp)
+
+if(TARGET Genivi::dlt)
+  target_compile_definitions(routingmanagerd PRIVATE USE_DLT)
+  target_link_libraries(routingmanagerd PRIVATE Genivi::dlt)
+endif()
+
 target_link_libraries(
   routingmanagerd
-  ${VSOMEIP_NAME}
-  ${Boost_LIBRARIES}
-  ${DL_LIBRARY}
-  ${DLT_LIBRARIES}
-  ${CMAKE_THREAD_LIBS_INIT}
+  PRIVATE vsomeip3
+          Boost::system
+          Boost::filesystem
+          STACKTRACE_INTERFACE
```

Comment:
could this also be an interface something like BOOST_INTERFACE, that would encompass every lib related to boost?

Reply:
Do you mean `Boost::system` and `Booot::filesystem`, or the `STACKTRACE_INTERFACE`?

Yeah, I think that could work.  Want me to do this?  It would likely reduce complexity (reduce the number of links to most targets)